src/main/scala-2.11/com/linagora/gatling/imap/action/ExpungeAction.scala

```diff
@@ -0,0 +1,27 @@
+package com.linagora.gatling.imap.action
+
+import akka.actor.Props
+import com.linagora.gatling.imap.check.ImapCheck
+import com.linagora.gatling.imap.protocol.Command
+import io.gatling.commons.util.TimeHelper._
+import io.gatling.commons.validation.Validation
+import io.gatling.core.action.ValidatedActionActor
+import io.gatling.core.session._
+
+import scala.collection.immutable.Seq
+
+object ExpungeAction {
+  def props(imapContext: ImapActionContext, requestname: String, checks: Seq[ImapCheck]) =
+    Props(new ExpungeAction(imapContext, requestname, checks))
+}
+
+class ExpungeAction(val imapContext: ImapActionContext, val requestName: String, override val checks: Seq[ImapCheck]) extends ValidatedActionActor with ImapActionActor {
+
+  override protected def executeOrFail(session: Session): Validation[_] = {
+    val id: Long = session.userId
+    val handler = handleResponse(session, nowMillis)
+    sessions.tell(Command.Expunge(id.toString), handler)
+
+    StaticStringExpression("null")(session)
```

Comment:
So I deleted the comment at the bad place. It was:

I know that is ugly but I had to return something of type Validation[_], I could not found something better.
Any suggestion ?